packages/react-hooks/src/useBlockInterval.ts

```diff
@@ -2,48 +2,29 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import type { ApiPromise } from '@polkadot/api';
+import type { BabeGenesisConfiguration } from '@polkadot/types/interfaces/babe';
 
 import { useMemo } from 'react';
 
-import { BN, BN_THOUSAND, BN_TWO, bnMin } from '@polkadot/util';
+import { BN } from '@polkadot/util';
 
 import { createNamedHook } from './createNamedHook.js';
 import { useApi } from './useApi.js';
-import { A_DAY } from './useBlocksPerDays.js';
-
-// Some chains incorrectly use these, i.e. it is set to values such as 0 or even 2
-// Use a low minimum validity threshold to check these against
-const THRESHOLD = BN_THOUSAND.div(BN_TWO);
-const DEFAULT_TIME = new BN(6_000);
-
-function calcInterval (api: ApiPromise): BN {
-  return bnMin(A_DAY, (
-    // Babe, e.g. Relay chains (Substrate defaults)
-    api.consts.babe?.expectedBlockTime ||
-    // POW, eg. Kulupu
-    api.consts.difficulty?.targetBlockTime ||
-    // Subspace
-    api.consts.subspace?.expectedBlockTime || (
-      // Check against threshold to determine value validity
-      api.consts.timestamp?.minimumPeriod.gte(THRESHOLD)
-        // Default minimum period config
-        ? api.consts.timestamp.minimumPeriod.mul(BN_TWO)
-        : api.query.parachainSystem
-          // default guess for a parachain
-          ? api.consts.aura?.slotDuration ?? DEFAULT_TIME.mul(BN_TWO)
-          // default guess for others
-          : DEFAULT_TIME
-    )
-  ));
-}
+import { useCall } from './useCall.js';
 
 function useBlockIntervalImpl (apiOverride?: ApiPromise | null): BN {
   const { api } = useApi();
 
+  const currApi = apiOverride || api;
+  const blockTimeAura = useCall(currApi.call.auraApi?.slotDuration && currApi.call.auraApi.slotDuration, []);
+  const blockTimeBabe = useCall(currApi.call.babeApi?.configuration && currApi.call.babeApi.configuration, [], {
+    transform: (data: BabeGenesisConfiguration | undefined) => data?.slotDuration
+  });
+
   return useMemo(
-    () => calcInterval(apiOverride || api),
-    [api, apiOverride]
-  );
+    () => (blockTimeAura || blockTimeBabe) ?? new BN(6_000),
```

Comment:
I think it would also make more sense here to have it be like `(blockTimeAura || blockTimeBabe) ?? calcInterval(curApi), [api, apiOverride]`.

My initial thought is, not all chains configured to apps are required to use either babe, or aura AFAIU. For example this might end of breaking for `Subspace`, and `Kulupu`, as seen in `calcInterval`

Reply:
Done in [c3383e9](https://github.com/polkadot-js/apps/pull/10457/commits/c3383e9710d6a6783b758290dbec11df36e9cef3).